### PR TITLE
Ports export reports

### PR DESCRIPTION
**Pull request by Anewbe**
Ports https://github.com/VOREStation/VOREStation/pull/2712

### CODE REVIEW

File: code/game/supplyshuttle.dm [hunk 1]
```diff
@@ -1,315 +1,336 @@
...
+	var/ordernum
+	var/list/shoppinglist = list()
+	var/list/requestlist = list()
+	var/list/supply_packs = list()
+	var/list/exported_crates = list()
+	//shuttle movement
+	var/movetime = 1200
+	var/datum/shuttle/ferry/supply/shuttle
+
+	New()
```
Comment (Atermonera) [inline code/game/supplyshuttle.dm]: Pathing

File: code/game/supplyshuttle.dm [hunk 2]
```diff
@@ -1,315 +1,336 @@
...
+		ordernum = rand(1,9000)
+
+		for(var/typepath in (typesof(/datum/supply_packs) - /datum/supply_packs))
+			var/datum/supply_packs/P = new typepath()
+			supply_packs[P.name] = P
+
+	// Supply shuttle ticker - handles supply point regeneration
+	// This is called by the process scheduler every thirty seconds
+	proc/process()
```
Comment (Atermonera) [inline code/game/supplyshuttle.dm]: Pathing

File: code/game/supplyshuttle.dm [hunk 3]
```diff
@@ -1,315 +1,336 @@
...
+		points += points_per_process
+
+	//To stop things being sent to CentCom which should not be sent to centcomm. Recursively checks for these types.
+	proc/forbidden_atoms_check(atom/A)
```
Comment (Atermonera) [inline code/game/supplyshuttle.dm]: Pathing

File: code/game/supplyshuttle.dm [hunk 4]
```diff
@@ -1,315 +1,336 @@
...
+		if(istype(A,/obj/item/device/radio/beacon))
+			return 1
+
+		for(var/i=1, i<=A.contents.len, i++)
+			var/atom/B = A.contents[i]
+			if(.(B))
+				return 1
+
+	//Sellin
+	proc/sell()
```
Comment (Atermonera) [inline code/game/supplyshuttle.dm]: Pathing

File: code/game/supplyshuttle.dm [hunk 5]
```diff
@@ -1,315 +1,336 @@
...
+			points += phoron_count * points_per_phoron
+
+		if(plat_count)
+			points += plat_count * points_per_platinum
+
+		if(money_count)
+			points += money_count * points_per_money
+
+	//Buyin
+	proc/buy()
```
Comment (Atermonera) [inline code/game/supplyshuttle.dm]: Pathing

**Comment by Anewbe** [comment]
> Morally Obligated

You know me better than that, Ater.

**Comment by KasparoVy** [comment]
That's a lot of porting going on.

**Comment by Atermonera** [comment]
Because anewbe's lazy, and also left a `world <<` in there

**Pull request merged**